### ISSUE TITLE
tests: coverage: test code coverage report failed on mps2_an385

### ISSUE THREAD

**Issue by enjiamai** [zephyrproject-rtos/zephyr#41388]
**Describe the bug**
Test code coverage report failed on mps2_an385. Now the report is blocked. There are also several test cases failed. 

1. The test case net/lib/coap/ will block the report, a PR https://github.com/zephyrproject-rtos/zephyr/pull/41340 is summited to fix this.

2. Most of them are due to stack size not enough, especially the test cases in tests/net. One problem here is the stack size we enlarged for gcov coverage in boards/arm/mps2_an385/Kconfig.defconfig, will be overlapped by prj.conf or testcase.yaml.
I thought there are three possible way to deal with it:

(1) Enlarge the demand stack size in all test cases of tests/net, but I think the stack size in these prj files. Take tests/net/lib/tls_credentials/ as example
```
diff --git a/tests/net/lib/tls_credentials/prj.conf b/tests/net/lib/tls_credentials/prj.conf
index 91424b1715..a53f0584f7 100644
--- a/tests/net/lib/tls_credentials/prj.conf
+++ b/tests/net/lib/tls_credentials/prj.conf
@@ -4,4 +4,4 @@ CONFIG_NETWORKING=y
 CONFIG_TEST_RANDOM_GENERATOR=y
 CONFIG_TLS_CREDENTIALS=y
 CONFIG_TLS_MAX_CREDENTIALS_NUMBER=4
-CONFIG_MAIN_STACK_SIZE=2048
+CONFIG_MAIN_STACK_SIZE=4096
```

(2) Add another testing scenario "coverage" for all the tests/net. But I really don't recommend this because this increases our number of testing being built.
```
diff --git a/tests/net/lib/tls_credentials/testcase.yaml b/tests/net/lib/tls_credentials/testcase.yaml
index cc8dae8a92..d5a45cba97 100644
--- a/tests/net/lib/tls_credentials/testcase.yaml
+++ b/tests/net/lib/tls_credentials/testcase.yaml
@@ -1,4 +1,13 @@
 tests:
   net.tls_credentials:
+    filter: not CONFIG_COVERAGE
     tags: net tls
     depends_on: netif
+    extra_configs:
+    - CONFIG_MAIN_STACK_SIZE=2048
+
+  net.tls_credentials.coverage:
+    filter: CONFIG_COVERAGE
+    depends_on: netif
+    extra_configs:
+    - CONFIG_MAIN_STACK_SIZE=4096
```

(3) Don't fix it.  Just run coverage report with following command to get the coverage report:
```
twister -p mps2_an385 -T tests/net/ -x=CONFIG_MAIN_STACK_SIZE=4096 -x=CONFIG_NET_TCP_WORKQ_STACK_SIZE=2048 -x=CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=4096
```
The drawback here is the stack size using might be large, if we want it suitable for all the testing.

So I need some help / suggestion for how I could fix it, thank for your help.

**To Reproduce**
1. source zephyr-env.sh
2. twister -T tests/ -p mps2_an385 --coverage
3. See error log

**Expected behavior**
The code coverage report can be generated successfully and test cases will pass the testing.

**Impact**
The code coverage report for mps2_an385 is blocked, when applying all testing.

**Logs and console output**
This is a example executing by "twister -T tests/net/lib/tls_credentials/ -p mps2_an385 --coverage"
```
E: ***** MPU FAULT *****
E:   Stacking error (context area might be not valid)
E:   Data Access Violation
E:   MMFAR Address: 0x2002e330
E: r0/a1:  0x00000000  r1/a2:  0x00000000  r2/a3:  0x00000000
E: r3/a4:  0x00000000 r12/ip:  0x00000000 r14/lr:  0x00000000
E:  xpsr:  0x00000000
E: Faulting instruction address (r15/pc): 0x00000000
E: >>> ZEPHYR FATAL ERROR 2: Stack overflow on CPU 0
E: Current thread: 0x2002b590 (main)
E: Halting system
~
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain : Zephyr SDK 0.13.1
 - Commit SHA or Version:  latest commit.

**Comment by enjiamai** [comment]
> How common is the issue and how generic the required stack changes are? If this is a common issue and if there can be found numbers for stack sizes that are needed for majority of builds maybe we can think about an option 4: Add somewhere an overlay called e.g. `coverage.conf` which will hold the required stack increases like the one from point (3) of enjiamai issue. Then, we modify twister in a way that whenever it is called with `--coverage` it will automatically use this overlay for all the builds.

@PerMac I like this idea, this is only use for code coverage. And we don't need to increase too much overlap files. Just a extra coverage.conf for special purpose. We can think about how to implement that.

**Comment by enjiamai** [comment]
Excuse me, I tend to close this because of following two reasons:
(1) The blocked test coverage report has been solved previously by #41340, we can run it without being blocked. (Though many test cases still failed while without extra needed stack size specified)
(2) A better way to address the stack size need to be more generic , so I filed another issue #44062 for that.

Thanks!
